warehouse/query-core/src/main/java/datawave/query/tables/RangeStreamScanner.java

```diff
@@ -646,4 +874,27 @@ private boolean isBeyondRange(Key lastSeenKey, Key endKey) {
             return false;
         }
     }
+    
+    // Overloaded
+    public static Entry<Key,Value> trimTrailingUnderscore(Entry<Key,Value> entry) {
+        Key nextKey = trimTrailingUnderscore(entry.getKey());
+        return new AbstractMap.SimpleEntry<>(nextKey, entry.getValue());
+    }
+    
+    /**
+     * It may be possible that a trailing underscore is appended to a day range. Check for and remove any trailing underscores that exist.
```

Comment:
Is this a bug we are compensating for?  Can we not fix where we are failing to trim that in the first place?

Reply:
It was an artifact of the condensed code path. I would prefer to leave this as is and clean up at a later point but am happy to defer to you on this.